docs/adr/006_cups.md

```diff
@@ -0,0 +1,297 @@
+# 006. CUPS Protocol Implementation - Credential management
+
+**Feature**: [#391](https://github.com/Azure/iotedge-lorawan-starterkit/issues/391)
+
+**Date**: 19 November 2021
+
+**Author**: Daniele Antonio Maggio
+
+**Status**: Proposed
+
+__________
+
+## Overview
+
+LoRa Basics™ Station defines a CUPS protocol for providing updated LNS/CUPS credentials and generic update data binary to any Client connecting to a CUPS Server.
+
+More information on the protocol can be found [here](https://doc.sm.tc/station/cupsproto.html).
+
+## In-scope
+
+This document focuses on:
+
+- Defining a flow diagram for common credential management scenarios:
+  - First connection
+  - Certificate rotation process
+- Defining the changes needed in IoT Hub for handling Basics Station specific information
+- Defining the changes needed in 'Facade' Azure Function for handling updated credential retrieval
+- Defining the changes needed in LoRaWan Network Server for handling above mentioned flows
+- Defining the changes needed in LoRa Device Provisioning CLI for handling creation/update of concentrators with certificate support
+
+## Out-of-scope
+
+Out of scope for this document is:
+
+- Defining a flow diagram for firmware update process
+- Defining any change needed in IoT Hub / Azure Function / Network Server for handling firmware updates
+
+## Authentication mechanisms with LoRa Basics™ Station
+
+### Context
+
+To better understand how the CUPS Protocol should be implemented, it is needed to understand how Basics Station is handling authentication.
+
+LoRa Basics™ Station supports four different authentication modes:
+
+- No Authentication
+- TLS Server Authentication
+- TLS Server and Client Authentication
+- TLS Server Authentication and Client Token
+
+An authentication mode can be set by configuring some files in the device where Basics Station is going to be executed.
```

Comment:
[nit] maybe a link to the docs which files those are would be helpful for following the flow of the document